tests/framework/resources.py

```diff
@@ -428,6 +428,75 @@ def create_json(
         return datax
 
 
+class SnapshotCreate():
+    """Facility for sending snapshot instructions on the microvm."""
+
+    SNAPSHOT_CREATE_URL = 'snapshot/create'
+
+    def __init__(self, api_usocket_full_name, api_session):
+        """Specify the information needed for sending API requests."""
+        url_encoded_path = urllib.parse.quote_plus(api_usocket_full_name)
+        api_url = API_USOCKET_URL_PREFIX + url_encoded_path + '/'
+        self._snapshot_cfg_url = api_url + self.SNAPSHOT_CREATE_URL
+        self._api_session = api_session
+
+    def put(self, **args):
+        """Create a snapshot of the microvm."""
+        datax = self.create_json(**args)
+        return self._api_session.put(
+            "{}".format(self._snapshot_cfg_url),
+            json=datax
+        )
+
+    @staticmethod
+    def create_json(mem_file_path, snapshot_path, diff=False):
```

Comment:
What about version? Even though we're not testing that now, it would be good to add a `version=None` param.

Reply:
yup, will add it

Reply:
done